stable/jenkins/templates/_helpers.tpl

```diff
@@ -74,11 +74,11 @@ Returns configuration as code default config
 jenkins:
 {{- if eq .Values.master.enableXmlConfig false }}
   {{- $configScripts := toYaml .Values.master.JCasC.configScripts }}
-  {{- if not (contains "authorizationStrategy:" $configScripts) }}
+  {{- if and (.Values.master.JCasC.authorizationStrategy) (not (contains "authorizationStrategy:" $configScripts)) }}
   authorizationStrategy:
     {{- tpl .Values.master.JCasC.authorizationStrategy . | nindent 4 }}
   {{- end }}
-  {{- if not (contains "securityRealm:" $configScripts) }}
+  {{- if and (.Values.master.JCasC.securityRealm) (not (contains "securityRealm:" $configScripts)) }}
```

Comment:
this looks good. clearing the default values already disables UI security but it's better not to render them at all.

values:
```
master:
  JCasC:
    authorizationStrategy: ""
    securityRealm: ""
```
before:
```
data:
  jcasc-default-config.yaml: |-
    jenkins:
      authorizationStrategy:
        
      securityRealm:
        
      disableRememberMe: false
```
after:
```
data:
  jcasc-default-config.yaml: |-
    jenkins:
      disableRememberMe: false
```